core/unix/signal.c

```diff
@@ -1276,7 +1290,8 @@ signal_thread_exit(dcontext_t *dcontext, bool other_thread)
         }
         info->num_pending = 0;
     }
-    if (!other_thread)
+    /* If no detach flag is set, we assume that this thread is on its way to exit. */
```

Comment:
Please explain the reason (in the comment) for not restoring the mask: if I were reading this code with the existing comment I would assume it was just a performance optimization to skip the mask set and that I could remove this doing_detach check and it would still be correct.

Reply:
Done